src/postman/transformers/request.ts

```diff
@@ -0,0 +1,138 @@
+import {
+  HttpParamStyles,
+  IHttpHeaderParam,
+  IHttpOperationRequest,
+  IHttpOperationRequestBody,
+  IHttpPathParam,
+  IHttpQueryParam,
+  IMediaTypeContent,
+} from '@stoplight/types';
+import { JSONSchema4 } from 'json-schema';
+import { FormParam, Header, HeaderList, PropertyList, QueryParam, Request, RequestBody } from 'postman-collection';
+import * as toJsonSchema from 'to-json-schema';
+import * as typeIs from 'type-is';
+import { transformDescriptionDefinition, transformValueToSchema } from '../util';
+
+export function transformQueryParam(queryParam: QueryParam): IHttpQueryParam {
+  return {
+    name: queryParam.key || '',
+    style: HttpParamStyles.Form,
+    ...(queryParam.value ? transformValueToSchema(queryParam.value) : undefined),
+  };
+}
+
+export function transformHeader(header: Header): IHttpHeaderParam {
+  return {
+    name: header.key,
+    style: HttpParamStyles.Simple,
+    ...(header.value ? transformValueToSchema(header.value) : undefined),
+  };
+}
+
+export function transformPathParams(segments: string[]): IHttpPathParam[] {
+  return segments.reduce<IHttpHeaderParam[]>((params, segment) => {
+    if (segment.startsWith(':')) {
+      params.push({
+        name: segment.substring(1),
+        style: HttpParamStyles.Simple,
+      });
+    }
+
+    return params;
+  }, []);
+}
+
+export function transformBody(body: RequestBody, mediaType?: string): IHttpOperationRequestBody | undefined {
+  switch (body.mode) {
+    case 'raw':
+      if (!body.raw) return;
+      return { contents: [transformRawBody(body.raw, mediaType)] };
+
+    case 'formdata':
+      if (!body.formdata) return;
+      return {
+        contents: [transformParamsBody<FormParam>(body.formdata, mediaType || 'multipart/form-data')],
+      };
+
+    case 'urlencoded':
+      if (!body.urlencoded) return;
+      return {
+        contents: [transformParamsBody<QueryParam>(body.urlencoded, mediaType || 'application/x-www-form-urlencoded')],
+      };
+  }
+
+  return;
+}
+
+function transformRawBody(raw: string, mediaType: string = 'text/plain'): IMediaTypeContent {
+  if (typeIs.is(mediaType, ['application/json', 'application/*+json'])) {
+    try {
+      const parsed = JSON.parse(raw);
+
+      return {
+        mediaType,
+        examples: [
+          {
+            key: 'default',
+            value: parsed,
+          },
+        ],
+        schema: toJsonSchema(parsed) as JSONSchema4,
+      };
+    } catch (e) {
+      /* noop, move on.. */
+    }
+  }
+
+  return {
+    mediaType,
+    examples: [
+      {
+        key: 'default',
+        value: raw,
+      },
+    ],
+  };
+}
+
+function transformParamsBody<T extends FormParam | QueryParam>(
+  params: PropertyList<T>,
+  mediaType: string,
+): IMediaTypeContent {
+  return {
+    mediaType,
+    schema: {
+      type: 'object',
+      properties: params.all().reduce<NonNullable<JSONSchema4['properties']>>((props, item) => {
+        props[item.key || ''] = {
```

Comment:
I'm really not sure whether we should generate artificial ids. This case actually have no good solutions:
1) artificial: user won't understand that, will omit that, will be surprised (chance: always when empty key is used)
2) allow keys to be overwritten: user will be surprised that some params has not been migrated (chance: >1 keys are empty)
3) ignore those with empty keys: user loses data (chances: alway when an empty key is used)

In my opinion, it would be more perfect to do 3). However, it sounds more user-friendly to do 1) with sth link `_generated_beef11`.

Ok, I convinced myself that 1) is good ;)

Reply:
Done! Tell me WDYT about it?